e2e/api_test.go

```diff
@@ -425,6 +441,180 @@ func TestE2E_TaskEndpoints_Delete_Conflict(t *testing.T) {
 	assert.Equal(t, http.StatusNotFound, resp.StatusCode)
 }
 
+// TestE2E_TaskEndpoints_Create tests the create task endpoint. This
+// runs a Consul server and the CTS binary in daemon mode.
+//	POST /v1/tasks
+func TestE2E_TaskEndpoints_Create(t *testing.T) {
+	t.Parallel()
+	// Test creating a task
+	// 1. Start with a task
+	// 2. Create infrastructure change that would trigger new task
+	// 3. Create a new task
+	// 4. Check that the new task exists
+	// 5. Check that the task did not run
+	// 6. Make a service change to a service tracked by the new task, verify an event exists
+
+	srv := testutils.NewTestConsulServer(t, testutils.TestConsulServerConfig{
+		HTTPSRelPath: "../testutils",
+	})
+	defer srv.Stop()
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "create_task")
+	initialTaskName := "initial-task"
+	cts := ctsSetup(t, srv, tempDir,
+		moduleTaskConfig(initialTaskName, "./test_modules/local_instances_file"))
+
+	// Make a change that would trigger the new task, if it existed
+	taskName := "created-task"
+	serviceName := "testService"
+	service := testutil.TestService{
+		ID:      serviceName,
+		Name:    serviceName,
+		Address: "5.6.7.8",
+		Port:    8080,
+	}
+	testutils.RegisterConsulService(t, srv, service, defaultWaitForRegistration)
+
+	// Create task
+	u := fmt.Sprintf("http://localhost:%d/v1/tasks",
+		cts.Port())
+
+	createTaskRequest := fmt.Sprintf(createTestTaskTemplate, taskName, serviceName)
+
+	resp := testutils.RequestHTTP(t, http.MethodPost, u, createTaskRequest)
```

Comment:
this utils returns the response and expects the caller to close. you'll want to add a `defer resp.Body.Close()`. there's a couple others that below as well that I won't leave a comment on

Reply:
I also did this for the other tests in this file which weren't calling close on the response bodies